project/source/Scripts/Objects/SourceContainer.gd

```diff
@@ -57,10 +57,10 @@ func SetContainerType(new: ContainerType) -> void:
 	containerType = new
 	SetupVisual()
 
-func CheckContents(group: StringName) -> bool:
-	return contents.is_in_group(group)
+func CheckContents(group: StringName) -> Array[bool]:
```

Comment:
Changing 
```func CheckContents(group: StringName) -> Array[bool]:```

Reply:
Seems like changing this file and the one below makes some of the interactions work again like adding a tip to a pipette.

Reply:
It *does*, but I would like to avoid using raw `Array` as much as possible. It turns out this was a problem with the way Godot determines the type in the ternary `if` expression. If you do `[contents.is_in_group('group')] if contents else []`, it treats the empty array as being `Array` instead of `Array[bool]`. So it can just be fixed by using a whole if statement like

```gdscript
if contents:
    return [contents.is_in_group('group')]
else:
    return []
```

Since the return type of the function is `Array[bool]`, directly returning `[]` gives it the right type.